### ISSUE TITLE
Plesk+nginx+SSL+JXcore 0.3.0.5?

### ISSUE THREAD

**Issue by Shorty1968** [jxcore/jxcore#591]
Hello if I specify in my Plesk under JXcore 0.3.0.5 the path to my SSL certificates, can I get a 502 Bad Gateway / nginx to somehow resolve with an additional Anweissung for NGINX in Plesk?

**Comment by ktrzeciaknubisa** [comment]
Can you explain in another words, what exactly do you mean?

**Comment by Shorty1968** [comment]
you mean in another words geman?

**Comment by ktrzeciaknubisa** [comment]
English...

**Comment by Shorty1968** [comment]
m,y english ist bad therefore I use yes Google Translate.

**Comment by gopumon** [comment]
I got a 502 when I tried the SSL ceritificate with key received. The solution is to make your SSL key to RSA with `openssl rsa -in server.key -out server_new.key`

**Comment by ktrzeciaknubisa** [comment]
@gopumon ok, what if you would try to add e.g. `ssl_ciphers ALL;` to nginx directives and use it the previous way? Or maybe any other nginx [ssl related](http://nginx.org/en/docs/http/ngx_http_ssl_module.html) directives need to be added?

**Comment by gopumon** [comment]
Sorry I am unable to recreate the scenario. SSL working fine with RSA encrypted key.

**Comment by ktrzeciaknubisa** [comment]
@Shorty1968 Please try the solution given by @gopumon , and let us know.

**Comment by Shorty1968** [comment]
I'd like to do, but unfortunately I do not know exactly how I have to do this with openssl rsa -in server.key -out server_new.key?

Do I have openssl rsa -in server.key -out server_new.key run in Putty and I must server.key replaced by something else?

**Comment by ktrzeciaknubisa** [comment]
Yes, run it with Putty, and replace "server.key" with your certificate file name. Then you will get new file "server_new.key". After that you can use it in JXcore extension for Plesk (on domain configuration page) to paste in *SSL certificate key file* field.

**Comment by Shorty1968** [comment]
thanks i have do this and have Trouble like Putty.

root@k045:~# openssl rsa -in computer-base.key -out computer-base_new.key
Error opening Private Key computer-base.key
140584403658400:error:02001002:system library:fopen:No such file or directory:bss_file.c:398:fopen('computer-base.key','r')
140584403658400:error:20074002:BIO routines:FILE_CTRL:system lib:bss_file.c:400:
unable to load Private Key

**Comment by ktrzeciaknubisa** [comment]
First you need to go  into directory, where your "computer-base.key" exists. For example:

```bash
cd /some_folder/
```

**Comment by Shorty1968** [comment]
Sorry as I could myself have to remember, I have now done so, but it is still 502 Bad Gatway if I Activate SSL.

root @ K045: openssl rsa -in /var/www/vhosts/computer-base.net/httpdocs/certificates# computer base.key -out computer base_new.key
writing RSA key

**Comment by ktrzeciaknubisa** [comment]
Did you copy "computer base_new.key" to your domain's httpdocs folder and then added  "computer base_new.key" to SSL settings in JXcore extension?

**Comment by Shorty1968** [comment]
I have now done everything exactly, but the 502 Bad Gatewy will not go away.

**Comment by Shorty1968** [comment]
I get it with a hook at SSL simply not to run, I may need for modules and if so which ones and where can I get these?
